CMakeLists.txt

```diff
@@ -4,7 +4,7 @@ project(sc-machine)
 site_name(www.ostis.net)
 cmake_policy(SET CMP0048 NEW)
 
-set(SC_MACHINE_ROOT ${CMAKE_CURRENT_SOURCE_DIR})
+set(SC_MACHINE_ROOT ${CMAKE_CURRENT_LIST_DIR})
```

Comment:
every other place in sc-machine uses source dir, if the only reason reason for this is that include("sc-machine/CMakeLists.txt") will work then not ok, it can be added with add_subdirectory(sc-machine)

Reply:
Many projects use `include` to include sc-machine to cmake tree

Reply:
Okay, let's use add_subdirectory.